### PR TITLE
Log stacktrace when tsd crashes, if available

### LINKED ISSUE #137: Errors getting logged without call stack or context
In reproducing https://github.com/SamVerschueren/tsd/issues/136, what you actually see on the terminal is:

```
$ tsd
Cannot read properties of undefined (reading 'flags')
```

No call stack. It's quite confusing.

The culprit is logging only the `message` here: https://github.com/SamVerschueren/tsd/blob/354f77d97315045770c1a94a4f16ef507c865dbd/source/cli.ts#L30

Could we have, say, a message like `Error running tsd: ${error.stack || error.message}` please? I'd be happy to send a PR!

### PR DESCRIPTION
**Pull request by ghost91-**
Closes #137

### CODE REVIEW

File: source/cli.ts [hunk 1]
```diff
@@ -26,8 +26,15 @@ const cli = meow(`
...
-		if (error && typeof (error as Error).message === 'string') {
-			console.error((error as Error).message);
+		const potentialError = error as Error | undefined | null;
+		const errorMessage =
+			typeof potentialError?.stack === 'string' ?
+				potentialError.stack :
+				typeof potentialError?.message === 'string' ?
+					potentialError?.message :
+					undefined;
+		if (errorMessage) {
```
Comment (sindresorhus) [inline source/cli.ts:36]: I think the check could be simpler.

File: source/cli.ts [hunk 2]
```diff
@@ -26,8 +26,15 @@ const cli = meow(`
...
 	} catch (error: unknown) {
-		if (error && typeof (error as Error).message === 'string') {
-			console.error((error as Error).message);
+		const potentialError = error as Error | undefined | null;
+		const errorMessage =
+			typeof potentialError?.stack === 'string' ?
+				potentialError.stack :
+				typeof potentialError?.message === 'string' ?
+					potentialError?.message :
+					undefined;
```
Comment (sindresorhus) [inline source/cli.ts:35]: I think the check could be simpler.

Comment (ghost91-) [reply source/cli.ts:35]: How would you simplify it?

Comment (sindresorhus) [reply source/cli.ts:35]:
```js
const errorMessage = potentialError?.stack ?? potentialError?.message;
```

File: source/cli.ts [hunk 3]
```diff
@@ -26,8 +26,15 @@ const cli = meow(`
 			throw new Error(formatter(diagnostics));
 		}
 	} catch (error: unknown) {
-		if (error && typeof (error as Error).message === 'string') {
-			console.error((error as Error).message);
+		const potentialError = error as Error | undefined | null;
```
Comment (sindresorhus) [inline source/cli.ts:29]:
```suggestion
		const potentialError = error as Error | undefined;
```

**Pull request closed without merging**